pantheon-bundle/src/test/java/com/redhat/pantheon/model/api/SlingModelAdapterFactoryTest.java

```diff
@@ -0,0 +1,92 @@
+import com.redhat.pantheon.model.module.ModuleVersion;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.testing.mock.sling.junit5.SlingContext;
+import org.apache.sling.testing.mock.sling.junit5.SlingContextExtension;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Calendar;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@ExtendWith({SlingContextExtension.class, MockitoExtension.class})
+class SlingModelAdapterFactoryTest {
+
+    private final SlingContext slingContext = new SlingContext();
+
+    SlingModelAdapterFactory modelAdapterFactory = new SlingModelAdapterFactory();
+
+    @Test
+    void getAdapter() {
+        // Given
+        slingContext.build()
+                .resource("/content/module1/draft/metadata",
+                        "jcr:createdBy", "a-user",
+                        "jcr:created", Calendar.getInstance())
+                .commit();
+
+        // When
+        ModuleVersion adapter = modelAdapterFactory.getAdapter(slingContext.resourceResolver().getResource("/content/module1/draft"),
+                ModuleVersion.class);
+
+        // Then
+        assertNotNull(adapter);
+        assertEquals("a-user", adapter.metadata().get().createdBy().get());
+        assertNotNull(adapter.metadata().get().created().get());
+    }
+
+    // Make sure there are no regressions that make adapters unavailable
+    @Test
+    void availableAdapters() {
+        // Given
+        Resource mockResource = Mockito.mock(Resource.class);
+
+        // Then
+        assertNotNull(modelAdapterFactory.getAdapter(mockResource, SlingModel.class));
+        assertNotNull(modelAdapterFactory.getAdapter(mockResource, Module.class));
+        assertNotNull(modelAdapterFactory.getAdapter(mockResource, ModuleVersion.class));
+        assertNotNull(modelAdapterFactory.getAdapter(mockResource, Content.class));
+        assertNotNull(modelAdapterFactory.getAdapter(mockResource, Metadata.class));
+        assertNotNull(modelAdapterFactory.getAdapter(mockResource, ModuleLocale.class));
+        assertNotNull(modelAdapterFactory.getAdapter(mockResource, Product.class));
+        assertNotNull(modelAdapterFactory.getAdapter(mockResource, ProductVersion.class));
+        assertNotNull(modelAdapterFactory.getAdapter(mockResource, FileResource.class));
+
+    }
+
+    @Test
+    void getUnrecognizedAdapter() {
+        // Given
+        slingContext.build()
+                .resource("/content/module1",
+                        "jcr:createdBy", "a-user",
+                        "jcr:created", Calendar.getInstance())
+                .commit();
+
+        // When
+        String adapter = modelAdapterFactory.getAdapter(slingContext.resourceResolver().getResource("/content/module1"),
+                String.class);
+
+        // Then
+        assertNull(adapter);
```

Comment:
Hmm, interesting. I would have expected an exception to be thrown in this case. I see you have code in the Adapter Factory that explicitly checks this and provides null, rather than throwing an exception. Would it violate convention to throw an exception?

Reply:
I think it might violate convention. Per the javadoc on the parent interface:
```
@return The adapted object or <code>null</code> if this factory instance cannot adapt the object.
```

Reply:
Fair enough!